M365DSC Dev branch via Azure DevOps Pipeline fails to load

<!--
    Your feedback and support is greatly appreciated, thanks for contributing!

    ISSUE TITLE:
    Please prefix the issue title with the resource name, e.g.
    'ResourceName: Short description of my issue'

    ISSUE DESCRIPTION (this template):
    Please provide information regarding your issue under each header below.
    Write N/A under any headers that do not apply to your issue, or if the
    information is not available.

    NOTE! Sensitive information should be obfuscated.

    PLEASE KEEP THE HEADERS.

    You may remove this comment block, and the other comment blocks,
    but please keep the headers.
-->
#### Details of the scenario you tried and the problem that is occurring
When using Azure DevOps pipeline with the following config, the output fails

#### Verbose logs showing the problem
```
2020-07-10T18:35:16.0076188Z ##[section]Starting: PowerShell Script
2020-07-10T18:35:16.0196755Z ==============================================================================
2020-07-10T18:35:16.0197145Z Task         : PowerShell
2020-07-10T18:35:16.0197463Z Description  : Run a PowerShell script on Linux, macOS, or Windows
2020-07-10T18:35:16.0197750Z Version      : 2.170.1
2020-07-10T18:35:16.0198022Z Author       : Microsoft Corporation
2020-07-10T18:35:16.0198403Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
2020-07-10T18:35:16.0198823Z ==============================================================================
2020-07-10T18:35:20.2488409Z Generating script.
2020-07-10T18:35:20.2488895Z ========================== Starting Command Output ===========================
2020-07-10T18:35:20.2490563Z ##[command]"C:\windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'd:\a\_temp\562a5776-0cb8-4959-b679-80a5e5a90b99.ps1'"
2020-07-10T18:35:22.1742841Z WinRM service is already running on this machine.
2020-07-10T18:35:22.2291477Z WinRM is not set up to allow remote access to this machine for management.
2020-07-10T18:35:22.2292390Z The following changes must be made:
2020-07-10T18:35:22.2292783Z 
2020-07-10T18:35:22.2293279Z Create a WinRM listener on HTTP://* to accept WS-Man requests to any IP on this machine. 
2020-07-10T18:35:22.2293945Z Configure LocalAccountTokenFilterPolicy to grant administrative rights remotely to local users.
2020-07-10T18:35:22.2294331Z 
2020-07-10T18:35:22.2424788Z WinRM has been updated for remote management.
2020-07-10T18:35:22.2431406Z 
2020-07-10T18:35:22.2432688Z Created a WinRM listener on HTTP://* to accept WS-Man requests to any IP on this machine. 
2020-07-10T18:35:22.2433629Z Configured LocalAccountTokenFilterPolicy to grant administrative rights remotely to local users.
2020-07-10T18:36:45.7037349Z WARNING: The names of some imported commands from the module 'Microsoft.Online.SharePoint.PowerShell' include 
2020-07-10T18:36:45.7038573Z unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the 
2020-07-10T18:36:45.7039490Z Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
2020-07-10T18:36:45.8747438Z WARNING: The names of some imported commands from the module 'Microsoft.PowerApps.Administration.PowerShell' include 
2020-07-10T18:36:45.8748603Z unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the 
2020-07-10T18:36:45.8750256Z Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
2020-07-10T18:36:46.8091024Z WARNING: The names of some imported commands from the module 'SharePointPnPPowerShellOnline' include unapproved verbs 
2020-07-10T18:36:46.8091832Z that might make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again
2020-07-10T18:36:46.8092296Z  with the Verbose parameter. For a list of approved verbs, type Get-Verb.
2020-07-10T18:38:03.6779912Z WARNING: The version '2.0.2.102' of module 'AzureADPreview' is currently in use. Retry the operation after closing the 
2020-07-10T18:38:03.6781747Z applications.
2020-07-10T18:38:17.6627720Z Import-Module : An instance of GraphSession already exists. Call Initialize(Func<GraphSession>, bool) to overwrite it.
2020-07-10T18:38:17.6631140Z At C:\Program 
2020-07-10T18:38:17.6631789Z Files\WindowsPowerShell\Modules\Microsoft.Graph.Authentication\0.7.0\Microsoft.Graph.Authentication.psm1:13 char:9
2020-07-10T18:38:17.6632889Z + $null = Import-Module -Name (Join-Path $PSScriptRoot '.\bin\Microsoft ...
2020-07-10T18:38:17.6633357Z +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2020-07-10T18:38:17.6633855Z     + CategoryInfo          : NotSpecified: (:) [Import-Module], InvalidOperationException
2020-07-10T18:38:17.6634364Z     + FullyQualifiedErrorId : System.InvalidOperationException,Microsoft.PowerShell.Commands.ImportModuleCommand
2020-07-10T18:38:17.6636283Z  
2020-07-10T18:38:17.7767609Z ##[error]PowerShell exited with code '1'.
2020-07-10T18:38:17.8285173Z ##[section]Finishing: PowerShell Script

```

#### Suggested solution to the issue

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)

```
# Configure the Agent to run DSC configurations
winrm quickconfig -force 

# Install the M365DSC module
Install-Module -Name Microsoft365DSC -Force   

#Install the Dev M365 DSC module
Install-M365DSCDevBranch

# Execute the DSC Configuration script from the built artefacts
&$(System.DefaultWorkingDirectory)/_M365-As-Code-CI/DSC/Config/M365_Config.ps1 

```


#### The operating system the target node is running
<!--
    Please provide as much as possible about the target node, for example
    edition, version, build and language.
    On OS with WMF 5.1 the following command can help get this information.

    Get-ComputerInfo -Property @(
        'OsName',
        'OsOperatingSystemSKU',
        'OSArchitecture',
        'WindowsVersion',
        'WindowsBuildLabEx',
        'OsLanguage',
        'OsMuiLanguages')
-->
Microsoft hosted agent "windows-2019"

#### Version of the DSC module that was used ('dev' if using current dev branch)
dev

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,9 @@
 
 ## UNRELEASED
 
+* MISC
+  * Renamed the event log to M365DSC to avoid journal conflicts;
+
 ## 1.20.716.1
 
 * MISC

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@
   * Updated Microsoft.Graph.Authentication module version to 0.7.1;
   * Updated Microsoft.PowerApps.Administration.PowerShell version to
     2.0.72;
+  * Updated SharePointPnPPowerShellOnline version to 3.23.2007.1;
 
 ## 1.20.716.1
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,8 +2,16 @@
 
 ## UNRELEASED
 
+* SPOAPP
+  * Added property Path as a key for the SPOApp resource to
+    prevent conflict where two solutions could have the same
+    name;
 * MISC
   * Renamed the event log to M365DSC to avoid journal conflicts;
+* Metadata
+  * Updated Microsoft.Graph.Authentication module version to 0.7.1;
+  * Updated Microsoft.PowerApps.Administration.PowerShell version to
+    2.0.72;
 
 ## 1.20.716.1
 

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_SPOApp/MSFT_SPOApp.schema.mof
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_SPOApp/MSFT_SPOApp.schema.mof
@@ -2,7 +2,7 @@
 class MSFT_SPOApp : OMI_BaseResource
 {
     [Key, Description("The name of the App.")] string Identity;
-    [Required, Description("The path the the app package on disk.")] string Path;
+    [Key, Description("The path the the app package on disk.")] string Path;
     [Write, Description("This will deploy/trust an app into the app catalog.")] boolean Publish;
     [Write, Description("Overwrites the existing app package if it already exists.")] boolean Overwrite;
     [Write, Description("Present ensures the site collection exists, absent ensures it is removed"), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;

--- a/Modules/Microsoft365DSC/Microsoft365DSC.psd1
+++ b/Modules/Microsoft365DSC/Microsoft365DSC.psd1
@@ -65,7 +65,7 @@
         },
         @{
             ModuleName      = "Microsoft.Graph.Authentication"
-            RequiredVersion = "0.7.0"
+            RequiredVersion = "0.7.1"
         },
         @{
             ModuleName      = "Microsoft.Graph.Groups.Planner"

--- a/Modules/Microsoft365DSC/Microsoft365DSC.psd1
+++ b/Modules/Microsoft365DSC/Microsoft365DSC.psd1
@@ -81,7 +81,7 @@
         },
         @{
             ModuleName      = "Microsoft.PowerApps.Administration.PowerShell"
-            RequiredVersion = "2.0.70"
+            RequiredVersion = "2.0.72"
         },
         @{
             ModuleName      = "MicrosoftTeams"
@@ -97,7 +97,7 @@
         },
         @{
             ModuleName      = "SharePointPnPPowerShellOnline"
-            RequiredVersion = "3.23.2007.0"
+            RequiredVersion = "3.23.2007.1"
         }
     )
 

--- a/Modules/Microsoft365DSC/Modules/M365DSCLogEngine.psm1
+++ b/Modules/Microsoft365DSC/Modules/M365DSCLogEngine.psm1
@@ -80,7 +80,7 @@ function Add-M365DSCEvent
         $EventID = 1
     )
 
-    $LogName = 'Microsoft365DSC'
+    $LogName = 'M365DSC'
 
     if ([System.Diagnostics.EventLog]::SourceExists($Source))
     {

